mapillary_tools/video_data_extraction/extract_video_data.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import typing as T
+from multiprocessing import Pool
+from pathlib import Path
+
+import tqdm
+
+import mapillary_tools.geotag.utils as video_utils
+
+from mapillary_tools import exceptions, geo, utils
+from mapillary_tools.geotag import gpmf_gps_filter
+from mapillary_tools.types import (
+    ErrorMetadata,
+    FileType,
+    MetadataOrError,
+    VideoMetadata,
+    VideoMetadataOrError,
+)
+from mapillary_tools.video_data_extraction.options import Options
+from mapillary_tools.video_data_extraction.video_data_parser_factory import make_parsers
+
+
+LOG = logging.getLogger(__name__)
+
+
+class VideoDataExtractor:
+    options: Options
+
+    def __init__(self, options: Options) -> None:
+        self.options = options
+
+    def process(self) -> T.List[MetadataOrError]:
+        files = utils.find_videos(self.options["paths"])
+        self._screen_sources_cardinality(files)
+        num_processes = self.options["num_processes"] or None
+
+        with Pool(processes=num_processes) as pool:
+            if num_processes == 1:
+                iter: T.Iterator[VideoMetadataOrError] = map(self.process_file, files)
+            else:
+                iter = pool.imap(self.process_file, files)
+
+            video_metadata_or_errors = list(
+                tqdm.tqdm(
+                    iter,
+                    desc="Extracting GPS tracks",
+                    unit="videos",
+                    disable=LOG.getEffectiveLevel() <= logging.DEBUG,
+                    total=len(files),
+                )
+            )
+
+        return video_metadata_or_errors
+
+    def process_file(self, file: Path) -> VideoMetadataOrError:
+        parsers = make_parsers(file, self.options)
+        points: T.Sequence[geo.Point] = []
+        make = None
+        model = None
+
+        exceptions = []
+        for parser in parsers:
+            log_details = {
+                "filename": file,
+                "parser": parser.parser_label,
+                "source": parser.get_geotag_source_path(),
+            }
+            if not points:
+                try:
+                    points = parser.extract_points()
+                except Exception as e:
+                    exceptions.append({"exception": e})
+                    LOG.debug(
```

Comment:
LOG as warning here perhaps? I assume it'd be worthy some attention if it throws.